README.md

```diff
@@ -37,11 +37,17 @@ A sample query : `/api/v1/search/bing?query=fossasia&format=xml`
 
 ## Dependencies
 
-* Python 2.x or Python 3.x
-* [Node.js](https://nodejs.org/en/)
-* [Pip](https://pip.pypa.io/en/stable/installing/)
-* [Flask](http://flask.pocoo.org/)
-* [BeautifulSoup4](https://www.crummy.com/software/BeautifulSoup/bs4/doc/)
+* [MongoDB](https://www.mongodb.com)
+* [Python 2.7 or Python 3.x](https://python.org)
+    * [Pip](https://pip.pypa.io/en/stable/installing)
+    * [BeautifulSoup4](https://www.crummy.com/software/BeautifulSoup/bs4/doc)
```

Comment:
Bs4 versions are different across Python2.x/3.x, we will need to handle that port separately, once the pending pulls are merged.

Reply:
Is there a todo here in README.md?  Please explain a bit more because it is not clear to me why we would use different versions of bs4 for Python 2 and Python 3.  In other projects, I use [the current bs4](https://pypi.python.org/pypi/beautifulsoup4) in both Python 2 and Python 3 without noticing any incompatibilities.

Reply:
There are certain discrepancies between the `bs4` versions for different versions of Python. Initially, up until most PRs don't get merged we would want to maintain cross compatibility across Python2.x/3.x. Now beautiful soup 4 has been updated to be compatible across the board, but some people who haven't run `pip install --upgrade beautifulsoup4` would face issues if they were using Python3.5 (check [this](https://stackoverflow.com/questions/28745153/importing-bs4-in-python-3-5))

Reply:
This project's [requirements.txt file](https://github.com/fossasia/query-server/commits/master/requirements.txt) has been specifying `beautifulsoup4>=4.5.1` since December 2016.  This version avoids this issue however if users install packages by hand and do not keep them up to date then they may have trouble.

We could put a link to `requirements.txt` in README.md if you believe that would clarify.

Reply:
Yeah, I think the best practice would be to have users explicitly install requirements through the `.txt` file. Maybe add a disclaimer to the README as well? In addition to this however, modules like librabbitmq (as stated by you) are Python3 incompatible, so we need some provision for the same as well.

Reply:
@Pk13055 says that we have to explicitly write out the best practice so I add it and then @vaibhavsingh97 wants me to take the best practice back out. Please get together and decide which way you want it!